### ISSUE TITLE
Activating brings white page

### ISSUE THREAD

**Issue by tyler71** [s22-tech/Yourls-Email-Notify#2]
Running YOURLS v 1.7.9& 

When clicking on activate in manage plugins, or following a short link (when activated) a white page is displayed
Refreshing the page brings back the admin menu

Commenting out parts of code, as far as I can tell these code blocks caused it to go white when activating.

```
   if ($code === 'xxx' || $is_bot === 'yes') exit;
```
```
   while ($result1 = $statement_1->fetch_object()) {                          
      $clicks     = $result1->clicks;                                         
      $title      = $result1->title;                                          
      $timestamp  = $result1->timestamp;                                      
   }
```

I've set the `$path` to a absolute directory -> `/home/myuser/websites/domain` with the directory existing but no error logs were created

**Comment by s22-tech** [comment]
Thank you so much for reporting this.  To fix, simply replace this line:
if ($code === 'xxx' || $is_bot === 'yes') exit;
with this:

if (isset($code) && ($code === 'xxx' || $is_bot === 'yes')) exit;

Marc

**Comment by s22-tech** [comment]
Actually, that fix won't work.  I tested it against version 1.8, which just came out today, and it still generated the white page.  I'll post a better fix as soon as I can, but for now just comment out or remove this line:

`if ($code === 'xxx' || $is_bot === 'yes') exit;`

**Comment by tyler71** [comment]
Ok,
I spun up a docker-compose demo instance to reproduce this issue so I'm happy to test any fixes.
Thank you for creating this!

**Comment by s22-tech** [comment]
I just uploaded a new version that should fix the problem.  Apparently 'exit' was the wrong choice for a plugin.  I've also verified that it works with version 1.8. 

Let me know how it works for you.

**Comment by tyler71** [comment]
I updated the file. Interestingly, while the docker demo one works fine now, my cpanel one does not.
I figured out how to enable php errors to be shown and I get this error on activation

```
Fatal error: Uncaught Error: Call to a member function fetch_object() on bool in /path/to/yourlsinstall/user/plugins/email-notify/plugin.php:154 
Stack trace: 
 #0 /path/to/yourlsinstall/includes/functions-plugins.php(152): s22_email_notification(Array) 
 #1 /path/to/yourlsinstall/includes/functions-plugins.php(191): yourls_apply_filter('pre_redirect', Array) 
 #2 /path/to/yourlsinstall/includes/functions.php(734): yourls_do_action('pre_redirect', 'https://domain.tld...', 302) 
 #3 /path/to/yourlsinstall/admin/plugins.php(27): yourls_redirect('https://domain.tld...', 302) 
 #4 {main} thrown in /path/to/yourlsinstall/user/plugins/email-notify/plugin.php on line 154
```
Your error log output
```
2021-02-22 09:09:54 -- my_ip_file: /path/to/yourlsinstall/data/files_to_watch/my_ip.txt
2021-02-22 09:09:54 -- args    : https://domain.tld/admin/plugins.php?success=deactivated,302
2021-02-22 09:09:54 -- long_url: https://domain.tld/admin/plugins.php?success=deactivated
2021-02-22 09:09:54 -- code    : xxx                                            
2021-02-22 09:09:54 -- keywords:                                                
2021-02-22 09:09:54 -- my_ip: x.x.x.x
```

___

I don't know php at all, but looking at this: https://stackoverflow.com/a/36148188/4172336
I modified it like so:


```
       while ($result1 = $statement_1->fetch_object()) {
->     while ($statement_1 && $result1 = $statement_1->fetch_object()) {
```

I am now able to activate / deactivate the plugin now

**Comment by s22-tech** [comment]
I had never tested this with an empty db before.  After reading your response, I tested it on a new install with an empty db but I didn’t get the fatal error that you did, so I don’t know what to tell you.  I will add the code you suggested, though, as I can see how it would help on a new installation.

**Comment by tyler71** [comment]
I edited the previous post, so this information may not of shown up by email
Tldr, cpanel installed the databases with its own table naming scheme, messing up the hardcoded table values. Updating the script fixed it

**Comment by s22-tech** [comment]
That's odd.  cPanel should not be messing with either your databases or your scripts, so it sounds like something else is going on.  Did it also change this line in /user/config.php?

`define( 'YOURLS_DB_PREFIX', 'yourls_' );`

If it didn’t, then the plugin should work with the hardcoded table names.

That being said, I found that there are constants for the table names, so I've used those instead which should alleviate any problems associated with hardcoding.  It also makes sense to use the constants as I didn’t realize before that the prefix could be changed in the config file.  This way, it will continue to work if someone does change it.

Thanks!

**Comment by s22-tech** [comment]
Made some major changes to the script.  I moved all the configurable variables out of the code and into the Click Notification Email admin page, so there's no reason to get into the code for those anymore.

Also fixed a few other problems.  The code that was mentioned from StackOverflow didn't actually work, so I found a fix for that, as well.

It should now for for anyone out-of-the-box.

**Comment by tyler71** [comment]
Works out of the box now 🎉
